### PR TITLE
basex: 8.6.6 -> 9.4.3

### PR DESCRIPTION
**Pull request by leenaars**
###### Motivation for this change

Version bump + refactoring

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by leenaars** [comment]
@bjornfor: I hope this version can be automatically updated.

**Review by bjornfor — APPROVED** [review]

File: pkgs/tools/text/xml/basex/default.nix [hunk 1]
```diff
@@ -25,23 +25,15 @@ stdenv.mkDerivation rec {
...
-
     # Remove Windows batch files (unclutter $out/bin)
-    rm -f "$out"/bin/*.bat
+    rm ./bin/*.bat
 
-    # Move some top-level stuff to $out/share/basex (unclutter $out)
-    mkdir -p "$out/share/basex"
-    mv "$out"/*.txt "$out/share/basex/"
-    mv "$out"/webapp "$out/share/basex/"
+    mkdir -p $out/share/basex" "$out/share/applications"
```
Comment (solson) [inline pkgs/tools/text/xml/basex/default.nix:31]: I cannot build the `basex` package on my machine because of the unmatched `"` in this line.

Comment (solson) [reply pkgs/tools/text/xml/basex/default.nix:31]: The details of the failed builds here show the same problem: 
image

**Comment by bjornfor** [comment]
Oops! Fixed in 7ad5e816faba3f808e2a8815b14d0f023a4e2160. Thanks, @solson!

**Comment by bjornfor** [comment]
Btw, do any of you want the newer version on release-20.09?

**Comment by solson** [comment]
I'm using `nixos-unstable` so no need for me.

**Pull request merged**